### PR TITLE
Lesson 12 Wiki Notes

### PR DESCRIPTION
**Pull request by gildedgardenia**

### CODE REVIEW

File: Class-12-(Model-Validation-and-Enums).md [hunk 1]
```diff
@@ -22,23 +23,25 @@
 
    b. How can bad data be dangerous?
 
+   c. Why would using a ViewModel be beneficial in model validation?
+
 1. Be sure that the students can differentiate between client and server side validation tools.
-1. Discuss some commonly annotations from the [csharp Validation API](https://csharpee.github.io/csharpee-spec/csharpdocs/csharpx/validation/constraints/package-summary.html).
-1. Walk the students through the *validation flow* mentioned in the text, perhaps with a new sample field on the ``Event`` model from ``coding-events`` .
+1. Discuss some commonly attributes from the [C#](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations?view=netcore-3.1).
```
Comment (CarlyLanglois) [inline Class-12-(Model-Validation-and-Enums).md]: this maybe a more appropriate link to guide this discussion: https://docs.microsoft.com/en-us/aspnet/core/mvc/models/validation?view=aspnetcore-3.1#built-in-attributes

File: Class-12-(Model-Validation-and-Enums).md [hunk 2]
```diff
@@ -22,23 +23,25 @@
...
+1. Walk the students through the *validation flow* mentioned in the text, perhaps with a new sample property on the ``Event`` model from ``CodingEvents`` .
```
Comment (CarlyLanglois) [inline Class-12-(Model-Validation-and-Enums).md]: this should be the viewmodel

File: Class-12-(Model-Validation-and-Enums).md [hunk 3]
```diff
@@ -22,23 +23,25 @@
...
 1. Compare strong error message and handling techniques to weaker examples.
 1. Address ``enum`` data types:
 
-   a. How to create an ``enum`` class and add custom properties.
+   a. How to create an ``enum`` and add custom properties.
```
Comment (CarlyLanglois) [inline Class-12-(Model-Validation-and-Enums).md]: enum properties arent really touched on in the text so lets ditch this part - i dont know if they exist the same way here as they do in java

**Review by CarlyLanglois — APPROVED** [review]

File: Class-12-(Model-Validation-and-Enums).md (hunk 1 above)
Comment (gildedgardenia) [reply Class-12-(Model-Validation-and-Enums).md]: Done!

File: Class-12-(Model-Validation-and-Enums).md (hunk 2 above)
Comment (gildedgardenia) [reply Class-12-(Model-Validation-and-Enums).md]: Done!

File: Class-12-(Model-Validation-and-Enums).md (hunk 3 above)
Comment (gildedgardenia) [reply Class-12-(Model-Validation-and-Enums).md]: Done!

**Pull request merged**